src/api/entities/Identity/index.ts

```diff
@@ -691,21 +706,44 @@ export class Identity extends Entity<UniqueIdentifiers, string> {
       },
     } = this;
 
-    const assembleResult = ({
-      secondaryKeys: secondaryAccounts,
-    }: PolymeshPrimitivesIdentity): PermissionedAccount[] => {
-      return secondaryAccounts.map(({ signer: rawSigner, permissions }) => ({
-        account: signatoryToAccount(rawSigner, context),
-        permissions: meshPermissionsToPermissions(permissions, context),
-      }));
+    const assembleResult = (
+      rawSecondaryKeyKeyRecord: Option<PolymeshPrimitivesSecondaryKeyKeyRecord>[],
+      accountIds: AccountId32[]
+    ): PermissionedAccount[] => {
+      return rawSecondaryKeyKeyRecord.reduce((secondaryKeys, optKeyRecord, index) => {
+        if (optKeyRecord.isSome) {
```

Comment:
This should always be Some, since the Identity is validated for existence when being fetched

Reply:
We should just unwrap an add a comment explaining why we know